internal/datasource/git.go

```diff
@@ -31,6 +32,17 @@ type GitSource struct{}
 
 func newGitSource() Sourcer { return &GitSource{} }
 
+func (s *GitSource) RefToOverride(ref *pb.Job_DataSource_Ref) (map[string]string, error) {
+	gitRef, ok := ref.Ref.(*pb.Job_DataSource_Ref_Git)
+	if !ok {
+		return nil, fmt.Errorf("ref is not a git ref: %T", ref.Ref)
```

Comment:
Does the `ref` here contain a proto status code that we should be passing along?